request_test.go

```diff
@@ -3991,3 +4002,52 @@ func TestRequest_Panics(t *testing.T) {
 		assert.Panics(t, func() { newRequest(newMockChain(t), config, "GET", "") })
 	})
 }
+
+func TestRequest_RetriesCustomFunc(t *testing.T) {
```

Comment:
I think we should cover for both
- `WithCustomRetryFunc` that returns `true`
- `WithCustomRetryFunc` that returns `false`

Wdyt?